### PR TITLE
Make TableView smoothier

### PR DESCRIPTION
**Pull request by boutinb**
smoothie

### CODE REVIEW

File: JASP-Desktop/components/JASP/Controls/TableView.qml [hunk 1]
```diff
@@ -212,42 +183,16 @@ JASPControl
 						anchors.topMargin:		- 8 * preferencesModel.uiScale
 						fieldHeight:			parent.height + 16 * preferencesModel.uiScale
 						fieldWidth:				parent.width
-						visible:				false
+						visible:				true
 						useExternalBorder:		false
 						value:					itemText
 						useLastValidValue:		false
+						focus:					true
```
Comment (JorisGoosen) [inline JASP-Desktop/components/JASP/Controls/TableView.qml]: why this?
it's a delegate and now they all get the focus set...

File: JASP-Desktop/qquick/datasetview.cpp [hunk 2]
```diff
@@ -78,10 +78,15 @@ void DataSetView::setRolenames()
...
 }
 
-void DataSetView::modelDataChanged(const QModelIndex &, const QModelIndex &, const QVector<int> &)
+void DataSetView::modelDataChanged(const QModelIndex &index, const QModelIndex &, const QVector<int> &)
 {
-	modelAboutToBeReset();
-	calculateCellSizes();
+	QSizeF calcedSize = getTextSize(_model->data(index).toString());
+
+	if (_cellSizes[size_t(index.column())].width() < calcedSize.width())
```
Comment (JorisGoosen) [inline JASP-Desktop/qquick/datasetview.cpp]: Now you are only comparing the current size of the cell with the new size of the cell. But the actual comparison should be made on the basis of the width of the whole column. 

Like: `_model->headerData(col, Qt::Orientation::Horizontal, _roleNameToRole["maxColString"]).toString();`
If you put that in a function you could reuse that easily, and then also use it at line 147

File: JASP-Desktop/qquick/datasetview.cpp [hunk 3]
```diff
@@ -137,14 +142,11 @@ void DataSetView::calculateCellSizes()
 	_colXPositions.resize(_model->columnCount());
 	_cellTextItems.clear();
 
-	_metricsFont = QFontMetricsF(_font);
-
 	for(int col=0; col<_model->columnCount(); col++)
 	{
 		QString text = _model->headerData(col, Qt::Orientation::Horizontal, _roleNameToRole["maxColString"]).toString();
```
Comment (JorisGoosen) [inline JASP-Desktop/qquick/datasetview.cpp]: Here you can use that function ^^

File: JASP-Desktop/qquick/datasetview.cpp [hunk 4]
```diff
@@ -539,7 +541,10 @@ void DataSetView::storeTextItem(int row, int col, bool cleanUp)
 
 	textItem->item->setVisible(false);
 
-	_textItemStorage.push(textItem);
+	if (_cacheItems)
```
Comment (JorisGoosen) [inline JASP-Desktop/qquick/datasetview.cpp]:
```
if (_cacheItems)	_textItemStorage.push(textItem);
else				delete textItem;
```
? ;)

File: JASP-Desktop/qquick/datasetview.cpp (hunk 2 above)
Comment (boutinb) [reply JASP-Desktop/qquick/datasetview.cpp]: OK?

**Pull request merged**